core/templates/dev/head/domain/question/QuestionPlayerBackendAPIService.js

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/**
+ * @fileoverview Service to receive questions for practice given a set of
+ * skill_ids.
+ */
+oppia.constant(
+  'QUESTION_PLAYER_URL_TEMPLATE',
+  '/question_player_handler?skill_ids=<skill_ids>&question_count' +
+  '=<question_count>&start_cursor=<start_cursor>');
+
+oppia.factory('QuestionPlayerBackendApiService', [
+  '$http', '$q', 'UrlInterpolationService', 'QUESTION_PLAYER_URL_TEMPLATE',
+  function($http, $q, UrlInterpolationService, QUESTION_PLAYER_URL_TEMPLATE) {
+    var _startCursor = '';
+    var _fetchQuestions = function(
+        skillIds, questionCount, successCallback, errorCallback) {
+      if (!_inputValidation(skillIds, questionCount, errorCallback)) {
+        return;
+      }
+      var questionDataUrl = UrlInterpolationService.interpolateUrl(
+        QUESTION_PLAYER_URL_TEMPLATE, {
+          skill_ids: skillIds,
+          question_count: questionCount,
+          start_cursor: _startCursor
+        });
+
+      $http.get(questionDataUrl).then(function(response) {
+        var questionDicts = angular.copy(response.data.question_dicts);
+        _startCursor = response.data.next_start_cursor;
+        if (successCallback) {
+          successCallback(questionDicts);
+        }
+      }, function(errorResponse) {
+        if (errorCallback) {
+          errorCallback(errorResponse.data);
+        }
+      });
+    };
+
+    /**
+     * Does basic validation on input
+     */
+    var _inputValidation = function(skillIds, questionCount, errorCallback) {
+      if (!skillIds) {
+        errorCallback('Skill ids value is expected but is missing');
+        return false;
+      }
+
+      var skillIdsList = skillIds.split(',');
```

Comment:
Why are skillIds passed in such a string format. Can't we pass them in as an array?
If strings are needed to have it in the URL, convert it to a string here or above instead of passing it around as comma separated values.

Reply:
to answer that question it is because `UrlInterpolationService` does not take an array for arguments. So we have to pass it CSV. 

A question here:
Will the frontend call `QuestionPlayerBackendApiService` with an array/list of skillIds or will it be a csv. If it is an array, then I will have to change the code to pass a csv to `UrlInterpolationService` to get the right url. If the frontend passes a csv, then no changes are needed.
cc: @vinitamurthi

Reply:
I can pass it as a csv that's fine. We can specify that its a requirement. If we feel that passing an array is better, we can do that later. I don't see a clear benefit of array over csv..what do you think Akshay?

Reply:
@seanlip regarding converting list to string for skillIds, if it is required that we will always get a string csv for skillIds, then we don't need to do that right?

Reply:
We should always convert explicitly and not use implicit conversion. This is actually only going to become more important, since we're currently working on introducing TypeScript into the codebase which adds more strictness to types.